plugins/azuread/pkg/azureclient/credential.go

```diff
@@ -68,14 +69,29 @@ func (c *RefreshTokenCredential) GetToken(ctx context.Context, options policy.To
 
 	// check for error
 	if responseData["error_description"] != nil {
-		errorMessage := responseData["error_description"].(string)
+		errorMessage, ok := responseData["error_description"].(string)
+		if !ok {
+			return accessToken, status.Error(codes.InvalidArgument, "failed to get error description")
+		}
+
 		return accessToken, status.Error(codes.InvalidArgument, errorMessage)
 	}
 
 	// retrieve the access token and expiration
-	accessToken.Token = responseData["access_token"].(string)
-	expiresIn := int(responseData["expires_in"].(float64))
-	accessToken.ExpiresOn = time.Now().Add(time.Second * time.Duration(expiresIn))
+	// retrieve the access token and expiration
+	token, ok := responseData["access_token"].(string)
+	if !ok {
+		return accessToken, status.Error(codes.InvalidArgument, "failed to cast access token to string")
```

Comment:
The error message is too low-level to be actionable for users. I would say `"'access_token' must be a string"`.